api/src/main/java/ai/djl/ndarray/BaseNDManager.java

```diff
@@ -156,7 +157,19 @@ public NDArray randomNormal(float loc, float scale, Shape shape, DataType dataTy
     /** {@inheritDoc} */
     @Override
     public NDArray truncatedNormal(float loc, float scale, Shape shape, DataType dataType) {
-        throw new UnsupportedOperationException("Not supported!");
+        int sampleSize = (int) shape.size();
+        double[] dist = new double[sampleSize];
+
+        for (int i = 0; i < sampleSize; i++) {
+            double sample = RandomUtils.nextGaussian();
+            while (sample < -2 || sample > 2) {
+                sample = RandomUtils.nextGaussian();
+            }
+
+            dist[i] = sample;
+        }
+
+        return create(dist).addi(loc).muli(scale).reshape(shape).toType(dataType, false);
```

Comment:
@AzizZayed I think you need to multiply before you add. Otherwise, it will scale the addition too.